internal/functions/deploy/bundle.go

```
@@ -60,7 +60,9 @@ func (b *dockerBundler) Bundle(ctx context.Context, slug, entrypoint, importMap
 	if viper.GetBool("DEBUG") {
 		cmd = append(cmd, "--verbose")
 	}
-	cmd = append(cmd, function.BundleFlags...)
+	if utils.Config.EdgeRuntime.DenoVersion == 1 {
+		cmd = append(cmd, function.BundleFlags...)
+	}
```

How would this work for the native bundler when we upgrade branching to edge runtime 2?

---

@sweatybridge the decorator flag no longer exists in edge runtime 2. These properties will be available to set in deno.json.

deno.json
```jsonc
{
  "compilerOptions": {
    // These options enable the experimental TypeScript decorators feature.
    // If this is not enabled, the default is tc39. (It was also the default in edge runtime 1)
    "experimentalDecorators": true,
    "emitDecoratorMetadata": false
  }
}
```

---

Can we just silently ignore this flag on edge runtime 2 rather than erroring out?

Then we don't have to worry about the upgrade path potentially breaking anyone.

---

Umm, I think that also will work. I'll do that. 😋